internal/verifier/change_stream.go

```diff
@@ -102,20 +125,38 @@ func (verifier *Verifier) iterateChangeStream(ctx context.Context, cs *mongo.Cha
 		return err
 	}
 
-	readOneChangeEvent := func() (bool, error) {
-		gotEvent := cs.TryNext(ctx)
-		if gotEvent {
-			var changeEvent ParsedEvent
-			if err := cs.Decode(&changeEvent); err != nil {
-				return false, errors.Wrap(err, "failed to decode change event")
+	readAndHandleOneChangeEventBatch := func() (bool, error) {
+		eventsRead := 0
+		var changeEventBatch []ParsedEvent
+
+		for hasEventInBatch := true; hasEventInBatch; hasEventInBatch = cs.RemainingBatchLength() > 0 {
+			gotEvent := cs.TryNext(ctx)
+
+			if !gotEvent || cs.Err() != nil {
+				break
+			}
+
+			if changeEventBatch == nil {
+				changeEventBatch = make([]ParsedEvent, cs.RemainingBatchLength()+1)
 			}
-			err := verifier.HandleChangeStreamEvent(ctx, &changeEvent)
-			if err != nil {
-				return false, errors.Wrap(err, "failed to handle change event")
+
+			if err := cs.Decode(&changeEventBatch[eventsRead]); err != nil {
+				return false, errors.Wrap(err, "failed to decode change event")
 			}
+
+			eventsRead++
+		}
+
+		if eventsRead > 0 {
+			verifier.logger.Debug().Msgf("Received a batch of %d events", eventsRead)
```

Comment:
Do you also want to log the 0 events case? I don't have enough context to know if that would be useful.

Reply:
I'd prefer not to log if there's no event. We've recommended users to run the m-v with debug log level, so I don't want too many log lines even at debug level.

Reply:
I don’t think 0 events is even possible since we wouldn’t be here unless TryNext() returned truthy.